src/interfaces/chain.h

```diff
@@ -245,8 +228,38 @@ class Chain
         virtual void chainStateFlushed(const CBlockLocator& locator) {}
     };
 
-    //! Register handler for notifications.
-    virtual std::unique_ptr<Handler> handleNotifications(std::shared_ptr<Notifications> notifications) = 0;
+    using ScanFn = std::function<Optional<uint256>(const uint256& start_hash, int start_height, const uint256& tip_hash, int tip_height)>;
+    using MempoolFn = std::function<void(std::vector<CTransactionRef>)>;
+
+    //! Register handler for notifications. Before returning and before sending
+    //! the first notification, call ScanFn with a scan range of blocks after a
+    //! specified location and time, and then call MempoolFn with a snapshot of
+    //! transactions in the mempool before the first transactionAddedToMempool /
```

Comment:
This part of comment may be made more intuitively instead of one chunk, like "Call MempoolFn with a snapshot of transaction before firing transactions back to caller".